src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/TimerTests.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Runtime.InteropServices.JavaScript.Tests
+{
+    // V8's implementation of setTimer ignores delay parameter and always run immediately. So it could not be used to test this.
+    [ConditionalClass(typeof(PlatformDetection), nameof(PlatformDetection.IsBrowserDomSupported))]
+    public class TimerTests : IAsyncLifetime
+    {
+        static JSObject _timersHelper;
+        static Function _installWrapper;
+        static Function _getRegisterCount;
+        static Function _getHitCount;
+        static Function _cleanupWrapper;
+        // static Function _log;
+
+        public static IEnumerable<object[]> TestCases()
+        {
+            yield return new object[] { new int[0], 0, null, null };
+            yield return new object[] { new[] { 10 }, 1 , null, null };
+            yield return new object[] { new[] { 10, 5 }, 2 , null, null };
+            yield return new object[] { new[] { 10, 20 }, 1 , null, null };
+            yield return new object[] { new[] { 800, 600, 400, 200, 050 }, 5, 13, 9 };
+        }
+
+        [Theory]
+        [MemberData(nameof(TestCases))]
+        public async Task TestTimers(int[] timeouts, int? expectedSetCounter, int? expectedSetCounterAfterCleanUp, int? expectedHitCount)
+        {
+            int wasCalled = 0;
+            Timer[] timers = new Timer[timeouts.Length];
+            try
+            {
+                // _log.Call(null, $"Waiting for runtime to settle");
```

Comment:
I would suggest uncommenting all of these, then changing them into _log?.Call so that you can turn this debugging stuff on/off all at once by assigning a value to _log

Reply:
done